Apromore-Boot/src/main/resources/menu.json

```diff
@@ -0,0 +1,93 @@
+{
+  "groups": [
+    {
+      "id": "FILE",
+      "items": [
+        {
+          "id": "UPLOAD"
+        },
+        {
+          "id": "DOWNLOAD"
+        },
+        {
+          "id": "SEPARATOR"
+        },
+        {
+          "id": "ETL"
+        },
+        {
+          "id": "JOB_SCHEDULER"
+        },
+        {
+          "id": "SEPARATOR"
+        },
+        {
+          "id": "CUT"
+        },
+        {
+          "id": "COPY"
+        },
+        {
+          "id": "PASTE"
+        },
+        {
+          "id": "SEPARATOR"
+        },
+        {
+          "id": "CREATE_FOLDER"
+        },
+        {
+          "id": "RENAME"
+        },
+        {
+          "id": "DELETE"
+        }
+      ]
+    },
+    {
+      "id": "DISCOVER",
+      "items": [
+        {
+          "id": "CREATE_MODEL"
+        },
+        {
+          "id": "DISCOVER_MODEL"
+        },
+        {
+          "id": "EDIT_MODEL"
+        },
+        {
+          "id": "FILTER_LOG"
+        }
```

Comment:
This menu.json is a good refactoring step, it can be use further to remove further the current complex dependencies:
- Add the group label and menu item label to menu.json.
For example: 
{
  "id": "DISCOVER",
  "label": "Discover" (or label code to support internationalization),
   "items": [
       {
            "id": "CREATE_MODEL",
            "label": "Create Model" (or label code to support internationalization)
       }
   ]  
}
- Remove getGroup and getLabel in PortalPlugin
- Removal getGroup, getLabel, getGroupLabel in each concrete PortalPlugin (e.g. CalendarPlugin, PD, etc.). They are decided based on menu.json
- the item id such as "id": "CREATE_MODEL" can be made more generic by using the plugin class canonical name for id, e.g. "id" : "org.apromore.plugin.portal.file.CreateProcessModelPlugin". This will help further remove the getID() method implementation in each concrete plugin like CreateProcessModelPlugin, just use one default implementation in PortalPlugin for getID(), e.g. this.getClass().getCanonicalName().